### PR TITLE
10540 - okta update login changes

### LINKED ISSUE #10540: Make content/design changes as needed to facilitate a better Okta migration for users
## Problem statement

Design and Content have created changes that need to be made to the website and Okta flows to facilitate a positive transition for users to the new Okta instance.

## What you need to know

This will be based on the change requests coming from Design and Content in these tickets:
- (#10537) Revise content of initial email to users
- (#10538) Revise content of login page to have better instructions for users
- (#10539) Revise design (if needed) of login page to have better instructions for users
- (#10541) Identify any changes/needs to the design/branding on rest of Okta instance needed for initial migration

Summary from Jillian:

- [User flow](https://www.figma.com/file/Du7feM8huRDAAZCQcUIj93/Website-Redesign?type=design&node-id=3191%3A695&mode=design&t=qkDTGOuOcCy422W2-1) with design and content updates
- [Cheat sheet](https://www.figma.com/file/Du7feM8huRDAAZCQcUIj93/Website-Redesign?type=design&node-id=3191%3A1280&mode=design&t=qkDTGOuOcCy422W2-1) with global design styles

## Acceptance criteria

- [ ] Content and components implemented and published
- [x] Update the asset logo in the Okta theme (get from Jillian)
- [x] Update the theme panel with new colors
- [ ] Move legal notice from the alert to the body of the page Login Page
- [ ] Repurpose the alert at the top of the page with the new content on the Login Page (DO NOT DEPLOY UNTIL 9/28 Deploy - captured in ticket #11340)
- [x] Update the content on the password reset alternate flow email
- [x] Update the content on the password reset normal flow email
- [x] Update the content of the automated activation email
- [x] Update content in the "Unable to Sign in" alert in the widget
- [x] Update remaining content per Figma file
- [ ] Update the ReportStream support page per Content update (#11069)

## To do

- [ ] ...

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

- Update content in the "Unable to Sign in" alert in the widget
- Update login content per Figma file
- OktaSignInWidget refactored and moved to shared
- shared Alert norms update
- okta-signin-widget patch to render as `div` instead of `main`
- doc on patches

Test Steps:
1. Will need to follow flow [here](https://www.figma.com/file/Du7feM8huRDAAZCQcUIj93/Website-Redesign?node-id=3191%3A1157&mode=dev)

## Changes
- Updates to oktaConfig.ts
-
### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

## Linked Issues
- Fixes #10540

### CODE REVIEW

**Comment by audreykwr** [comment]
I think I got through everything except the normal password reset. Any screen with a note in the below list has a orange/yellow box that says AR flag in Figma, which may help your reference.

- [ ] "Sign In" on button needs to change to "Sign in"
- [ ] "Forgot password?" now needs to say "Reset password"
- [ ] Odd line break behavior (cutting words) in the alerts under entering username but I don't know if that's something we can fix. I can share a screenshot in Slack
- [ ] Password reset screen - "Reset password" needs to be in sentence case
- [ ] Subject of alternate reset email flow is not "account password reset but" "ReportStream account password reset requested"
- [ ] Email subject of activation email needs to be "Finish setting up your ReportStream account"
- [ ] MFA set up - were we able to change anything about that flow? The desire was for ReportStream to replace "your company" at the top, and for the buttons to be changed to "set up" from "setup" but maybe this was one of those things we couldn't change?
- [ ] I didn't see an updated token expiration screen when I tried to use my token again. It was still the original
- [ ] Normal reset password flow - sentence case Reset password and Username or email
- [ ] Lockout alert needs to read: "Your account is locked because of too many failed attempts. Check your email for next steps to unlock."
- [ ] Change account locked email subject and header to "ReportStream account locked"
- [ ] Remove the Okta help url at the bottom of the account locked email
- [ ] Change unlock account email header and subject to: "ReportStream account unlock requested"

On support page:
- [ ] The question needs to change from "How do I reset my password" to  "How do I update or reset my password?" That will change on the card at the top as well as the actual question

File: frontend-react/docs/patches.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Patches
```
Comment (etanb) [inline frontend-react/docs/patches.md:1]: I think this piece of information should be in our `README`?

**Review by etanb — APPROVED** [review]
Just a small question re: documentation.

File: frontend-react/docs/patches.md (hunk 1 above)
Comment (jpandersen87) [reply frontend-react/docs/patches.md:1]: As a link, for sure, but I don't think it should clutter the main readme.

**Comment by audreykwr** [comment]
- [ ] Subject of alternate reset email flow is not "account password reset but" "ReportStream account password reset requested"

This was checked but the email I just got did not reflect this change

**Review by etanb — APPROVED** [review]

**Comment by audreykwr** [comment]
- [ ] Check password reset emails are both titled: "ReportStream account password reset requested"

**Comment by audreykwr** [comment]
- [ ] The title of both password reset email subject is correct but the title still needs to be "ReportStream account password reset requested" Let me know if you want me to screenshot the Figma reference for additional context.

- [ ] Sentence case "Email or username" for password reset screen

Screenshot 2023-10-16 at 8 51 50 PM

**Comment by audreykwr** [comment]
Another thing - The spacing on the login page banner gets wonky when windows resize. I'm not sure if that's just a component issue though.

**Review by audreykwr — APPROVED** [review]

**Review by jillian-hammer — APPROVED** [review]

File: frontend-react/src/App.tsx [hunk 2]
```diff
@@ -30,11 +131,15 @@ const App = ({ children }: AppProps) => {
             sessionTimeAggregate.current
         );
     };
+    const Fallback = useCallback(() => <ErrorPage type="page" />, []);
 
     useEffect(() => {
         const onUnload = () => {
-            trackAppInsightEvent(EventName.SESSION_DURATION, {
-                sessionLength: calculateAggregateTime() / 1000,
+            appInsights.trackEvent({
```
[the remark this answers is not in the document]
Comment (penny-lischer) [reply frontend-react/src/App.tsx:138]: Is it b/c you need the appInsights object later?

File: frontend-react/src/oktaConfig.ts [hunk 3]
```diff
@@ -18,30 +24,83 @@ const oktaAuthConfig: OktaAuthOptions = {
...
     logo: "/assets/cdc-logo.svg",
     language: "en",
     features: {
         registration: false, // Disable self-service registration flow
         rememberMe: false, // Setting to false will remove the checkbox to save username
-        router: true, // Leave this set to true for the API demo
-        // TODO: bring up enabling these once we have OktaPreview
-        // securityImage: true, // helps prevent spoofing, may require CORS to allow it through.
-        // mfaOnlyFlow: true, // government requires mfa
+        router: false, // Leave this set to true for the API demo
```
Comment (penny-lischer) [inline frontend-react/src/oktaConfig.ts:57]: Should this be true?

Comment (jpandersen87) [reply frontend-react/src/oktaConfig.ts:57]: With it enabled it changes the URL in the browser during the login process (/signin/*) which is a conflict with our app router. I haven't found any reason why we would want this.

Comment (penny-lischer) [reply frontend-react/src/oktaConfig.ts:57]: maybe add this comment to the code so that others understand better.

**Review by penny-lischer — COMMENTED** [review]
LGTM

**Comment by penny-lischer** [comment]
Still LGTM

File: frontend-react/src/hooks/UseCreateFetch.ts [hunk 4]
```diff
@@ -60,17 +62,18 @@ export const auxExports = {
...
-                oktaToken as Partial<AccessToken>,
+                authState?.accessToken as Partial<AccessToken>,
                 activeMembership as MembershipSettings,
+                fetchHeaders,
             ),
-        [oktaToken, activeMembership],
+        [authState?.accessToken, activeMembership, fetchHeaders],
     );
 
     return generator;
```
Comment (etanb) [inline frontend-react/src/hooks/UseCreateFetch.ts:79]: @jpandersen87 I logged in fine, but then once I went to `/admin/settings` I get:

Screenshot 2023-10-20 at 3 00 14 PM

**Review by etanb — CHANGES_REQUESTED** [review]

File: frontend-react/src/hooks/UseCreateFetch.ts (hunk 4 above)
Comment (penny-lischer) [reply frontend-react/src/hooks/UseCreateFetch.ts:79]: For me, I get the following:
Screenshot 2023-10-20 at 3 05 13 PM

**Comment by penny-lischer** [comment]
LGTM

**Review by etanb — APPROVED** [review]

**Comment by jpandersen87** [comment]
engineer reference of highlights of improvements while preparing frontend for migration: https://docs.google.com/document/d/1FXj4QDLXD4Ym0xYo7OoFxsarec3_g3URGkiURt_jYhc

**Pull request merged**